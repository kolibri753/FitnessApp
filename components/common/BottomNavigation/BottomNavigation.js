import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { colors } from '../../../styles/colors';
import styles from './BottomNavigation.styles';

const BottomNavigation = ({ state, descriptors, navigation }) => {
  return (
    <View style={styles.container}>
      {state.routes.map((route, index) => {
        const { options } = descriptors[route.key];
        const label = options.title || route.name;

        const isFocused = state.index === index;

        const onPress = () => {
          const event = navigation.emit({
            type: 'tabPress',
            target: route.key,
            canPreventDefault: true,
          });

          if (!isFocused && !event.defaultPrevented) {
            navigation.navigate(route.name);
          }
        };

        return (
          <TouchableOpacity
            key={route.key}
            onPress={onPress}
            style={[styles.tab, isFocused && styles.activeTab]}
          >
            <Ionicons name={options.iconName} size={24} color={isFocused ? colors.yellow : colors.white} />
            <Text style={[styles.tabLabel, isFocused && styles.activeTabLabel]}>{label}</Text>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default BottomNavigation;
