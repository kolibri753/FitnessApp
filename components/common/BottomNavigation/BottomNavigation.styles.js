import { StyleSheet } from 'react-native';
import { colors } from '../../../styles/colors';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: colors.black,
    height: 60,
    borderTopWidth: 1,
    borderTopColor: colors.lightGrey,
    paddingHorizontal: 20,
  },
  tab: {
    alignItems: 'center',
  },
  tabLabel: {
    fontSize: 12,
    color: colors.white,
  },
  activeTab: {
    borderBottomWidth: 2,
    borderBottomColor: colors.yellow,
  },
  activeTabLabel: {
    color: colors.yellow,
  },
});

export default styles;
