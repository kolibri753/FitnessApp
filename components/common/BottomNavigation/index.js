import BottomNavigation from './BottomNavigation';
import styles from './BottomNavigation.styles';

export { BottomNavigation, styles };