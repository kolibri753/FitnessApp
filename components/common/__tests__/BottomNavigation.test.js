import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import BottomNavigation from "../BottomNavigation";

describe("BottomNavigation", () => {
  test("renders correctly", () => {
    const state = {
      routes: [
        { key: "Home", name: "Home" },
        { key: "Exercises", name: "Exercises" },
        { key: "Workouts", name: "Workouts" },
        { key: "Profile", name: "Profile" },
      ],
      index: 0,
    };

    const descriptors = {
      Home: { options: { title: "Home" } },
      Exercises: { options: { title: "Exercises" } },
      Workouts: { options: { title: "Workouts" } },
      Profile: { options: { title: "Profile" } },
    };

    const navigation = {
      emit: jest.fn(),
      navigate: jest.fn(),
    };

    const { getByText } = render(
      <BottomNavigation
        state={state}
        descriptors={descriptors}
        navigation={navigation}
      />
    );

    expect(getByText("Home")).toBeTruthy();
    expect(getByText("Exercises")).toBeTruthy();
    expect(getByText("Workouts")).toBeTruthy();
    expect(getByText("Profile")).toBeTruthy();
  });

  test("calls navigation.navigate when a tab is pressed", () => {
    const state = {
      routes: [
        { key: "Home", name: "Home" },
        { key: "Exercises", name: "Exercises" },
        { key: "Workouts", name: "Workouts" },
        { key: "Profile", name: "Profile" },
      ],
      index: 0,
    };

    const descriptors = {
      Home: { options: { title: "Home" } },
      Exercises: { options: { title: "Exercises" } },
      Workouts: { options: { title: "Workouts" } },
      Profile: { options: { title: "Profile" } },
    };

    const navigation = {
      emit: jest.fn((eventName, eventPayload) => {
        // Mocking navigation.emit to return an object with defaultPrevented property
        return {
          defaultPrevented: false,
        };
      }),
      navigate: jest.fn(),
    };

    const { getByText } = render(
      <BottomNavigation
        state={state}
        descriptors={descriptors}
        navigation={navigation}
      />
    );

    fireEvent.press(getByText("Exercises"));
    expect(navigation.navigate).toHaveBeenCalledWith("Exercises");
  });

  it("renders correctly", () => {
    const state = {
      routes: [
        { key: "Home", name: "Home" },
        { key: "Exercises", name: "Exercises" },
        { key: "Workouts", name: "Workouts" },
        { key: "Profile", name: "Profile" },
      ],
      index: 0,
    };

    const descriptors = {
      Home: { options: { title: "Home" } },
      Exercises: { options: { title: "Exercises" } },
      Workouts: { options: { title: "Workouts" } },
      Profile: { options: { title: "Profile" } },
    };

    const navigation = {
      emit: jest.fn(),
      navigate: jest.fn(),
    };

    const tree = renderer
      .create(
        <BottomNavigation
          state={state}
          descriptors={descriptors}
          navigation={navigation}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
