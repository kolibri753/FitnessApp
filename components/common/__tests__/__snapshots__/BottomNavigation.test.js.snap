// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavigation renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderTopColor": "#eee",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-around",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#FFD500",
        "borderBottomWidth": 2,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFD500",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          },
          {
            "color": "#FFD500",
          },
        ]
      }
    >
      Home
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          },
          false,
        ]
      }
    >
      Exercises
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          },
          false,
        ]
      }
    >
      Workouts
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          },
          false,
        ]
      }
    >
      Profile
    </Text>
  </View>
</View>
`;
